src/vos/vos_tree.c

```diff
@@ -154,7 +154,9 @@ ktr_hkey_gen(struct btr_instance *tins, d_iov_t *key_iov, void *hkey)
 {
 	struct ktr_hkey		*kkey = (struct ktr_hkey *)hkey;
 	struct umem_pool        *umm_pool = tins->ti_umm.umm_pool;
+	struct vos_pool         *pool     = (struct vos_pool *)tins->ti_priv;
 
+	D_ASSERT(key_iov->iov_len < pool->vp_pool_df->pd_scm_sz);
```

Comment:
if we want to be really "safe", we could ensure it doesn't go off the end of the pool but let's see if this works for now to quiet coverity